CRM/Upgrade/Incremental/php/FourSeven.php

```diff
@@ -1232,6 +1233,37 @@ public static function pickActivityRevisionPolicy(CRM_Queue_TaskContext $ctx) {
   }
 
   /**
+   * Remove the contribution logging reports which have been broken for a very long time.
+   *
+   * @param \CRM_Queue_TaskContext $ctx
+   *
+   * @return bool
+   */
+  public static function removeContributionLoggingReports(CRM_Queue_TaskContext $ctx) {
+    if (class_exists('CRM_Report_Form_Contribute_LoggingDetail') || class_exists('CRM_Report_Form_Contribute_LoggingSummary')) {
+      // Perhaps the site has overridden these classes. The core ones are broken but they
+      // may have functional ones.
+      return TRUE;
+    }
+    $options = civicrm_api3('OptionValue', 'get', array('option_group_id' => 'report_template', 'options' => array('limit' => 0)));
+    foreach ($options['values'] as $option) {
+      $instances = civicrm_api3('ReportInstance', 'get', array('report_id' => $option['value']));
+      if ($instances['count']) {
+        foreach ($instances['values'] as $instance) {
+          if ($instance['navigation_id']) {
```

Comment:
shouldn't this be nested under the If in new Line 1259?

Reply:
hot diggity dog! you are right!

Reply:
fix pushed

Reply:
probably everything from L1250 to L1258 needs to be under the if